syntax = "proto3";

package grpc_cpp_server;

message RuleCreateRequest {
  string rule_str        = 1;
  uint64 rule_client_id  = 2;
}

message RuleCreateResponse {
  uint64 rule_client_id  = 1;
  uint64 rule_server_id  = 2;
}

message RuleDeleteRequest {
  uint64 rule_client_id = 1;
  uint64 rule_server_id = 2;
}

message RuleDeleteResponse {
  uint64 rule_client_id = 1;
  uint64 rule_server_id = 2;
}

service RuleManager {
  rpc StreamDataBidirectional1(stream RuleCreateRequest) returns (stream RuleCreateResponse);
  rpc StreamDataBidirectional2(stream RuleDeleteRequest) returns (stream RuleDeleteResponse);
}

